realm/realm-library/src/androidTestObjectServer/java/io/realm/SessionTests.java

```diff
@@ -52,4 +60,76 @@ public void get_syncValues() {
         assertEquals(user, session.getUser());
         assertEquals(configuration, session.getConfiguration());
     }
+
+    // Check that a Client Reset is correctly reported.
+    @Test
+    @RunTestInLooperThread
+    public void errorHandler_clientResetReported() {
+        SyncUser user = createTestUser();
+        String url = "realm://objectserver.realm.io/default";
+        final SyncConfiguration config = configFactory.createSyncConfigurationBuilder(user , url)
+                .errorHandler(new SyncSession.ErrorHandler() {
+                    @Override
+                    public void onError(SyncSession session, ObjectServerError error) {
+                        fail("Wrong error " + error.toString());
+                    }
+
+                    @Override
+                    public void onClientReset(SyncSession session, ClientResetRequiredError error) {
+                        String filePathFromError = error.getOriginalFile().getAbsolutePath();
+                        String filePathFromConfig = session.getConfiguration().getPath();
+                        assertEquals(filePathFromError, filePathFromConfig);
+                        assertFalse(error.getBackupFile().exists());
+                        assertTrue(error.getOriginalFile().exists());
+                        looperThread.testComplete();
+                    }
+                })
+                .build();
+
+        Realm realm = Realm.getInstance(config);
+        looperThread.testRealms.add(realm);
+
+        // Trigger error
+        SyncManager.simulateClientReset(SyncManager.getSession(config));
+    }
+
+    // Check that we can manually execute the Client Reset.
+    @Test
+    public void errorHandler_manualExecuteClientReset() {
+        SyncUser user = createTestUser();
+        String url = "realm://objectserver.realm.io/default";
+        final SyncConfiguration config = configFactory.createSyncConfigurationBuilder(user , url)
+                .errorHandler(new SyncSession.ErrorHandler() {
+                    @Override
+                    public void onError(SyncSession session, ObjectServerError error) {
+                        fail("Wrong error " + error.toString());
+                    }
+
+                    @Override
+                    public void onClientReset(SyncSession session, ClientResetRequiredError error) {
+                        try {
+                            error.executeClientReset();
```

Comment:
Yes, this is why I'm concerned about the naming of things. Perhaps the method should be `onClientResetRequired` instead, to make it more clear that when this method is called it means the Client should do a Reset, but one hasn't actually been done yet.